ucp/utils.py

```diff
@@ -75,10 +75,28 @@ def filter(self, record):
     return logger
 
 
+def _ensure_cuda_device(devs, rank):
+    import numba.cuda
+
+    dev_id = devs[rank % len(devs)]
+    os.environ["CUDA_VISIBLE_DEVICES"] = str(dev_id)
+    numba.cuda.current_context()
```

Comment:
It _should_ be sufficient to use `numba.cuda.select_device(dev_id)` without resorting to `os.environ` munging, but that needs https://github.com/NVIDIA/cuda-python/issues/24 to be fixed first.

Reply:
Yes, that's right. This isn't critical here as well, so let's wait for that and then we can do it properly once that's completed.